src/dataset.rs

```diff
@@ -70,12 +70,16 @@ impl<B: Backend> Batcher<FSRSItem, FSRSBatch<B>> for FSRSBatcher<B> {
                     item.history().map(|r| (r.delta_t, r.rating)).unzip();
                 delta_t.resize(pad_size, 0);
                 rating.resize(pad_size, 0);
-                let delta_t =
-                    Tensor::from_data(Data::new(delta_t, Shape { dims: [pad_size] }).convert())
-                        .unsqueeze();
-                let rating =
-                    Tensor::from_data(Data::new(rating, Shape { dims: [pad_size] }).convert())
-                        .unsqueeze();
+                let delta_t = Tensor::from_data(
+                    Data::new(delta_t, Shape { dims: [pad_size] }).convert(),
+                    &B::Device::default(),
```

Comment:
For FSRSBatcher and FSRS, you can use &self.device instead, which is a little cleaner, but I don't know if it's any more efficient.